### PR TITLE
docs(x/ecocredit): add basket and update content

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #833

Add basket documentation and update ecocredit core and marketplace documentation.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct `docs:` prefix in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the [documentation writing guidelines](https://github.com/cosmos/cosmos-sdk/blob/master/docs/DOC_WRITING_GUIDELINES.md)
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct `docs:` prefix in the PR title
- [ ] confirmed all author checklist items have been addressed 
- [ ] confirmed that this PR only changes documentation
- [ ] reviewed content for consistency
- [ ] reviewed content for thoroughness
- [ ] reviewed content for spelling and grammar
- [ ] tested instructions (if applicable)

### CODE REVIEW

File: x/ecocredit/spec/01_concepts.md [hunk 1]
```diff
@@ -1,87 +1,111 @@
...
+## Basket Submodule
 
-## Retired Credits
+### Basket
 
-Retired credits are credits that cannot be transferred between accounts nor can they be unretired. Retired credits are equivalent to burned tokens with the exception that retired credits are actively tracked after being retired. Retiring a credit implies that the holder of a credit is “claiming” the credit as an offset. Credits can be retired upon issuance, retired upon transfer, and retired by the owner of the credits. The retirement location is required upon retirement.
+A basket is an abstraction for different types of credits that meet a defined criteria. Credits can be put into a basket in exchange for an equivalent amount of basket tokens. The basket criteria can be set to only accept specific credit types, credit classes, and credit batches that meet a specific date criteria (e.g. credit batches with a minimum start date, credit batches with a start date within a duration of time, or credit batches with a start date year within a number of years into the past). Basket tokens can be returned to the basket at any time in exchange for the equivalent amount of credits.
```
Comment (technicallyty) [inline x/ecocredit/spec/01_concepts.md:73]: in the current impl, only one credit type can be specified 
```suggestion
A basket is an abstraction for different types of credits that meet a defined criteria. Credits can be put into a basket in exchange for an equivalent amount of basket tokens. The basket criteria can be set to only accept a specific credit type, credit classes, and credit batches that meet a specific date criteria (e.g. credit batches with a minimum start date, credit batches with a start date within a duration of time, or credit batches with a start date year within a number of years into the past). Basket tokens can be returned to the basket at any time in exchange for the equivalent amount of credits.
```

**Review by technicallyty — APPROVED** [review]

File: x/ecocredit/spec/01_concepts.md [hunk 2]
```diff
@@ -1,87 +1,111 @@
...
+Credit types are listed as an on-chain parameter. Adding a new credit type to the list of allowed credit types requires a parameter change proposal.
 
-## Credit Class Admin
+For more information about the properties of a credit type, see [CreditType](https://buf.build/regen/regen-ledger/docs/main/regen.ecocredit.v1#regen.ecocredit.v1.CreditType).
 
-The credit class admin is defined within a credit class as the address with the authority to manage and update the credit class. When a new credit class is created, the admin is always initially set to the address that created the credit class. The credit class admin will have the ability to transfer the admin role to another address, manage the list of credit class issuers, and change credit class metadata.
+### Credit Class Creator Allowlist
 
-## Credit Issuers
+The ecocredit module supports the option of restricting credit class creation to a set of addresses. The option to enable the credit class creator allowlist and the list of allowed credit class creators are both on-chain parameters that can only be updated through parameter change proposals.
```
Comment (clevinson) [inline x/ecocredit/spec/01_concepts.md:33]:
```suggestion
The ecocredit module supports the option of restricting credit class creation to a set of addresses. Enabling this restriction and maintaining the list of allowed credit class creators are both controlled via on-chain governance and can only be updated through parameter change proposals.
```

File: x/ecocredit/spec/01_concepts.md [hunk 3]
```diff
@@ -1,87 +1,111 @@
...
 
-The credit issuers are defined within a credit class as a list of addresses with the authority to mint new credits and issue credit batches of the corresponding credit class. The list of credit issuers are defined at the time the credit class is created. The credit class admin will be able to manage the list of credit issuers for the credit class that they administer.
+### Credit Class Admin
 
-## Credit Batch
+The credit class admin is defined within a credit class as the address with the authority to update the credit class (i.e. to update the admin, the list of approved issuers, and the metadata). When a credit class is created, the admin is initially set to the address that created the credit class.
```
Comment (clevinson) [inline x/ecocredit/spec/01_concepts.md:37]:
```suggestion
The credit class admin is the address with the authority to update a given credit class (i.e. to update the admin, the list of approved issuers, and the metadata). When a credit class is created, the admin is initially set to the address that created the credit class.
```

File: x/ecocredit/spec/01_concepts.md [hunk 4]
```diff
@@ -1,87 +1,111 @@
...
+
+### Credit Class Issuers
+
+The credit class issuers are defined within a credit class as the addresses with the authority to issue credit batches under the credit class. The list of credit class issuers is defined at the time the credit class is created and only the admin can update the list after the credit class is created.
```
Comment (clevinson) [inline x/ecocredit/spec/01_concepts.md:41]:
```suggestion
The issuers of a given credit class are the addresses with the authority to issue credit batches under that credit class. The list of credit class issuers is set at the time the credit class is created and only the admin can update the list after the credit class is created.
```

File: x/ecocredit/spec/01_concepts.md [hunk 5]
```diff
@@ -1,87 +1,111 @@
...
+
+### Credit Batch
 
 Credits are issued in batches by credit issuers granted the authority to issue credits for a given credit class. A credit batch refers to a batch of credits issued at a single point in time.
```
Comment (clevinson) [inline x/ecocredit/spec/01_concepts.md:45]:
```suggestion
All credits are issued in discrete batches by credit issuers. A credit batch refers to a batch of credits issued at a single point in time for a given credit class and project.
```

File: x/ecocredit/spec/01_concepts.md [hunk 6]
```diff
@@ -1,87 +1,111 @@
...
+
+For more information about the properties of an allowed denom, see [AllowedDenom](https://buf.build/regen/regen-ledger/docs/main/regen.ecocredit.marketplace.v1#regen.ecocredit.marketplace.v1.AllowedDenom).
+
+<br/>
 
 ---
 
+<br/>
+
 ![Ecocredit Types](./assets/types.png)
```
Comment (clevinson) [inline x/ecocredit/spec/01_concepts.md:109]: can this be moved to the top of this page?

File: x/ecocredit/spec/01_concepts.md [hunk 7]
```diff
@@ -1,87 +1,111 @@
...
 
 ![Credit Class Roles](./assets/roles.png)
```
Comment (clevinson) [inline x/ecocredit/spec/01_concepts.md:111]: can this be moved right under credit class admin & credit class issuers section? Maybe change those to subheadings, and call the parent section "Credit Class Roles" ?

Comment (clevinson) [reply x/ecocredit/spec/01_concepts.md:111]: Also maybe move allowlist (below) image to be right under the description of hte allow list.

Comment (clevinson) [reply x/ecocredit/spec/01_concepts.md:111]: Also maybe move allowlist (below) image to be right under the description of hte allow list.

**Review by clevinson — APPROVED** [review]
Pre approving with a handful of text suggestions.  Feel free to commit + merge as you see fit.

File: x/ecocredit/spec/01_concepts.md [hunk 8]
```diff
@@ -1,87 +1,111 @@
...
-- **issuer list**: The list of regen addresses who are allowed to issue credit batches under this credit class
-- **credit type**: The primary indicator for this credit class (e.g. Carbon measured in Tons Equiv. CO2 sequestered)
-- **metadata**: A byte array (up to 256 bytes) which can be used to store small amounts of metadata, or a URI that points to an off-chain resource for querying more complete metadata information. This usually would include descriptive information about the credit class's acceptable methodologies for monitoring changes in ecological state.
+### Credit Class
 
-Once a new credit class is created, credits can be issued at-will in distinct batches by any address in the **issuer list** of the corresponding credit class.
+A credit class is the primary abstraction for an ecosystem service credit. A credit class is associated with a credit type that represents the primary unit of measurement for the accepted methodologies included within the credit class. Information about the approved methodologies are typically stored off-chain, securely hashed, timestamped, and then included in the metadata field.
```
[the remark this answers is not in the document]
Comment (ryanchristo) [reply x/ecocredit/spec/01_concepts.md:7]: Slightly modified and accepted. For future reference, "off-chain data" (hyphen) and "stored off chain" (no hyphen) as well as "hash-based" (hyphen). I kept the link but we should update to point to concepts in the data module once we have them.

File: x/ecocredit/spec/01_concepts.md (hunk 6 above)
Comment (ryanchristo) [reply x/ecocredit/spec/01_concepts.md:109]: I actually think we should keep these at the bottom of the page for now as they don't align 1-to-1 with the concepts and they are becoming outdated. I was considering removing them but I think they still provide some value.

**Pull request merged**